adam/curriculum_to_html.py

```diff
@@ -0,0 +1,177 @@
+            # output_text.append(f"<li>Properties:\n<ul>")
+            # for prop in obj.properties:
+            #    output_text.append(f"<li>{prop.handle}</li>")
+            # output_text.append("</ul></li>")
+        output_text.append("\t\t\t</ul>")
+        if situation.actions:
+            output_text.append("\t\t\t<h4>Actions</h4>\n\t\t\t\t<ul>")
+            for acts in situation.actions:
+                output_text.append(f"\t\t\t\t<li>{acts.action_type.handle}</li>")
+            output_text.append("\t\t\t</ul>")
+        if situation.relations:
+            output_text.append("\t\t\t<h4>Relations</h4>\n\t\t\t<ul>")
+            for rel in situation.relations:
+                output_text.append(
+                    f"\t\t\t\t<li>{rel.relation_type.handle}({rel.first_slot.ontology_node.handle},{rel.second_slot.ontology_node.handle})</li>"
+                )
+            output_text.append("\t\t\t</ul>")
+        output_text.append("\t\t</td>")
+        return output_text
+
+    @staticmethod
+    def _perception_text(
+        perception: PerceptualRepresentation[DevelopmentalPrimitivePerceptionFrame]
+    ) -> Iterable[str]:
+        """
+        Turns a perception into a list of items in the perceptions frames.
+        """
+        output_text = [f"\t\t<td>"]
+        frame_number = 0
+        for frame in perception.frames:
+            output_text.append(f"\t\t\t<h4>Frame {frame_number}</h4>")
+            output_text.append("\t\t\t\t<h5>Perceived Objects</h5>\n\t\t\t\t<ul>")
+            for obj in frame.perceived_objects:
+                output_text.append(f"\t\t\t\t\t<li>{obj.debug_handle}</li>")
+            output_text.append("\t\t\t\t</ul>")
+            if frame.property_assertions:
+                output_text.append("\t\t\t\t<h5>Property Assertions</h5>\n\t\t\t\t<ul>")
+                for prop in frame.property_assertions:
+                    output_text.append(f"\t\t\t\t\t<li>{prop}")
+                output_text.append("\t\t\t\t</ul>")
+            if frame.relations:
+                output_text.append("\t\t\t\t<h5>Relations</h5>\n\t\t\t\t<ul>")
+                for rel in frame.relations:
+                    output_text.append(
+                        f"\t\t\t\t\t<li>{rel.relation_type.handle}({rel.arg1},{rel.arg2})"
+                    )
+                output_text.append("\t\t\t\t</ul>")
+            frame_number = frame_number + 1
+        output_text.append("\t\t</td>")
+        return output_text
+
+    @staticmethod
+    def _linguistic_text(linguistic: LinearizedDependencyTree) -> Iterable[str]:
+        """
+        Parses the Linguistic Description of a Linearized Dependency Tree into a table entry
+
+        Takes a `LinearizedDependencyTree` which is turned into a token sequence and
+        phrased as a sentence for display. Returns a List[str]
+        """
+        return [f"\t\t<td>", " ".join(linguistic.as_token_sequence()), "\t\t</td>"]
+
+
+def main(params: Parameters):
```

Comment:
You'll also need to call this from a

```
if __name__ == "__main__":
```

block for Python to run it.  You may want to use `parameters_only_entry_point` from vistautils.